dpgen2/exploration/report/report_trust_levels_base.py

```diff
@@ -180,27 +219,47 @@ def _record_one_traj(
         if novirial:
             assert id_v_accu is None
             assert id_v_fail is None
+        nomagforce = id_mf_cand is None
+        if nomagforce:
+            assert id_mf_accu is None
+            assert id_mf_fail is None
         nframes = np.size(np.concatenate((id_f_cand, id_f_accu, id_f_fail)))
         if (not novirial) and nframes != np.size(
             np.concatenate((id_v_cand, id_v_accu, id_v_fail))
         ):
             raise FatalError("number of frames by virial ")
+        if (not nomagforce) and nframes != np.size(
+            np.concatenate((id_mf_cand, id_mf_accu, id_mf_fail))
+        ):
+            raise FatalError("number of frames by magnetic force ")
         # nframes
         # to sets
+        set_full = set([ii for ii in range(nframes)])
         set_f_accu = set(id_f_accu)
         set_f_cand = set(id_f_cand)
         set_f_fail = set(id_f_fail)
-        set_v_accu = set([ii for ii in range(nframes)]) if novirial else set(id_v_accu)
+        set_v_accu = set_full if novirial else set(id_v_accu)
```

Comment:
dangerous to assign `set_full` to `set_v_acc`, as `set_v_acc` will be a reference to `set_full`, which would introduce some unexpected behavior. see the following code
```python
set_full = set([1,2,3])
set_v = set_full
set_mf = set_full
set_v.add(4)
print(set_mf)
```
The code gives 
```
{1, 2, 3, 4}
```